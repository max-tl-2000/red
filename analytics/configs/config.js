/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

module.exports = {
  analyticsJobs: [
    {
      jobGroup: 'RESTARTMART',
      fileName: 'L3_drop.sql',
      jobName: 'L3_DROP',
      sequenceNumber: 1,
    },
    {
      jobGroup: 'RESTARTMART',
      fileName: 'L3_DDL.sql',
      jobName: 'L3_DDL',
      sequenceNumber: 10,
    },
    {
      jobGroup: 'LOADMART',
      fileName: 'L3_load_dimensions.sql',
      jobName: 'L3_LOAD_DIMENSIONS',
      sequenceNumber: 80,
    },
    {
      jobGroup: 'LOADMART',
      fileName: 'L3_load_facts.sql',
      jobName: 'L3_LOAD_FACTS',
      sequenceNumber: 81,
    },
  ],
};
