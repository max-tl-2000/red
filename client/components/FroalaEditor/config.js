/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

export const editorDefaultFontFamily = 'sans-serif,Arial,Helvetica';
export const editorDefaultFontSize = '13';
const editorFontSizes = [editorDefaultFontSize, '18', '32'];

export const getConfig = ({ froalaEditor, toolbarId, height = null }) => ({
  fullPage: true,
  useClasses: false,
  toolbarBottom: true,
  listAdvancedTypes: false,
  quickInsertEnabled: false,
  emoticonsUseImage: false,
  linkAlwaysBlank: true,
  enter: froalaEditor.ENTER_DIV,
  imageUpload: false,
  toolbarContainer: `#${toolbarId}`,
  htmlAllowComments: true,
  iframeStyle: `body{ padding:0px; font-size:${editorDefaultFontSize}px; font-family:${editorDefaultFontFamily}; }`,
  fontFamily: {
    "serif,'Times New Roman',Times": 'Serif',
    editorDefaultFontFamily: 'Sans Serif',
  },
  fontSize: editorFontSizes,
  fontSizeDefaultSelection: editorDefaultFontSize,
  toolbarButtons: {
    moreText: {
      buttons: [
        'attachFile',
        'bold',
        'italic',
        'underline',
        'strikeThrough',
        'customFontSize',
        'fontFamily',
        'textColor',
        'backgroundColor',
        'pre',
        'clearFormatting',
      ],
      buttonsVisible: 1,
    },
    moreParagraph: {
      buttons: ['blockquote', 'alignLeft', 'alignCenter', 'formatUL', 'formatOLSimple'],
      buttonsVisible: 0,
    },
    moreRich: {
      buttons: ['emoticons', 'insertLink', 'insertHR'],
      buttonsVisible: 3,
    },
  },
  pluginsEnabled: [
    'codeView',
    'fontSize',
    'fontFamily',
    'colors',
    'quote',
    'paragraphFormat',
    'paragraphStyle',
    'draggable',
    'align',
    'link',
    'lists',
    'image',
    'emoticons',
    'url',
    'entities',
    'inlineClass',
    'inlineStyle',
  ],
  heightMin: height || 390,
  heightMax: height || 390,
  events: {
    keydown({ key, metaKey, ctrlKey }) {
      if (key === 'Enter' && (metaKey || ctrlKey)) {
        this.sendMessage();
      }
    },
  },
  iframeStyleFiles: ['https://cdnjs.cloudflare.com/ajax/libs/froala-editor/3.2.6/css/froala_style.min.css'],
  key: 'Ja2A4wA3C2E1E1C4B3nDc2YRTYKg1Dc2a1JVVG1VJKKYLMPvA1E1I1C2B8C7E7E1F5H5==',
  pastePlain: true,
});
