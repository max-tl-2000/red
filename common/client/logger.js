/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

import { window } from '../helpers/globals';
import { isString } from '../helpers/type-of';

const getBrowserData = () => {
  const { navigator, location } = window;

  const { userAgent } = navigator;
  const { href: currentUrl } = location;

  return { userAgent, currentUrl };
};

const logMethods = ['trace', 'debug', 'info', 'error', 'warn'];

// these will be printed to the console so we can check then in FS
const allowedLevels = ['info', 'error', 'warn'];

export const logger = logMethods.reduce((acc, methodName) => {
  acc[methodName] = (...args) => console.warn('logger not initialized', ...args);
  return acc;
}, {});

export const initClientLogger = ({ apiClient: client, getContextData } = {}) => {
  if (!client) {
    throw new Error('apiClient parameter not provided');
  }

  const logMessage = (level, logData, message) => {
    const logArgs = {
      loggingMessage: message,
      severity: level,
      ...getBrowserData(),
      ...logData,
    };

    if (getContextData) {
      logArgs.contextData = getContextData();
    }

    const shouldPostToServer = logData?.shouldPostToServer;

    if (level === 'error' || shouldPostToServer) {
      client
        .post('/log', {
          data: [logArgs],
        })
        .catch(err => {
          console.error('UNABLE TO POST LOG!', err);
        });
    }

    if (process.env.NODE_ENV === 'development') {
      console[level](message, logData);
    } else if (allowedLevels.includes(level)) {
      // in case of production like logs will be caught by FS
      // so we can see then also in the console during the session replies
      const { loggingMessage, ...argsForLogging } = logArgs;
      let obj = argsForLogging;

      try {
        obj = JSON.stringify(argsForLogging, null, 2);
        console[level](`${loggingMessage}, ${obj}`);
      } catch (err) {
        console[level](loggingMessage, obj);
      }
    }
  };

  return logMethods.reduce((acc, methodName) => {
    acc[methodName] = (logData, message = '') => {
      if (isString(logData)) {
        message = logData;
        logData = {};
      }

      logMessage(methodName, logData, message);
    };
    return acc;
  }, logger);
};
