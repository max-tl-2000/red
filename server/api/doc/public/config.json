{
  "swagger": "2.0",
  "info": {
    "description": "Move your app forward with the Reva API. In order to generate the api_key please perform a login with swagger and copy the returned token in the api_key textbox at the top.",
    "title": "Reva API",
    "version": "1.0.0"
  },

  "paths": {
    "/config": {
      "get": {
        "responses": {
          "200": {
            "description": "Tenant settings and config data"
          },
          "400": {
            "description": "Invalid/missing parameter",
            "schema": {
              "$ref": "../commons.json#/definitions/Error"
            }
          }
        },
        "summary": "Returns the tenant settings and config data",
        "tags": [
          "Config"
        ]
      }
    }
  }
}
