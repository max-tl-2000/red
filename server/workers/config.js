/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

import extend from 'extend';
import path from 'path';
import { createConfig } from '../../common/server/create-config';
import { createRentappConfig } from '../../rentapp/server/workers/create-rentapp-config';
import { cacheResults } from '../../common/server/cache-results';

// workers config is loading the rentapp/workers/config. Not entirely sure why this is done like this, proably we should just copy those configs over here
// but that will require tons of changes.

// For now applying this hack will make the code to work again. We need to refactor this at some point.
// config modules were intended to contain only Primitive values, having configs with functions and using ES6 Syntax with imports forces babel to traverse
// almost the entire set of modules from the very beginning. This process might take up to a minute (mostly due to this bug with the babel cache: https://github.com/babel/babel/issues/5667)
//
// TODO: Refactor the code to remove functions from configs and possibly remove this hack
const { __workerConfigFn: baseWorkersCfgFn, ...cfg } = createConfig({ configsDir: path.resolve(__dirname, './configs'), addWorkersConfigGetter: false });
const { __workerConfigFn: overrideWorkersCfgFn, ...rentappCfg } = createRentappConfig({ addWorkersConfigGetter: false });

const result = extend(true, cfg, rentappCfg, {
  RETRY_EXCHANGE: `${cfg.env}_retry_topic_exchange`,
  DEAD_LETTER_EXCHANGE: `${cfg.env}_dead_letter_topic_exchange`,
});

const workerCfgFn = cacheResults(() => {
  const base = baseWorkersCfgFn() || {};
  const overrider = overrideWorkersCfgFn() || {};

  return extend(true, base, overrider);
});

Object.defineProperty(result, 'workerConfig', {
  get() {
    return workerCfgFn();
  },
});

export default result;
