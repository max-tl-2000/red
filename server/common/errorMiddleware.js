/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

import config from '../config';
import logger from '../../common/helpers/logger';
import { USER_AUTHORIZATION_ERROR_TOKENS } from '../../common/enums/error-tokens';
import { verboseLogCtx } from './verboseLogCtx';
import { formatError } from '../../common/server/format-error';

const isLoginError = error => error && error.message && error.message in USER_AUTHORIZATION_ERROR_TOKENS;
const isInvalidJWTError = error => error && error.code && error.code.toUpperCase() === 'INVALID_TOKEN';
const isExpiredInvite = error => error.status === 498;

// eslint-disable-next-line no-unused-vars
export const errorHandler = () => (err, req, res, next) => {
  if (err && err.redirect) {
    res.redirect(err.redirect);
  } else {
    if (err.name === 'JsonSchemaValidation') {
      config.logMiddlewareErrors && logger.error({ error: err, ...verboseLogCtx(req), validations: err.validations }, 'JSONSchemaValidation');
      res.status(400);

      const responseData = {
        statusText: 'Bad Request',
        jsonSchemaValidation: true,
        validations: err.validations,
      };
      res.json(responseData);
      return;
    }
    if (config.logMiddlewareErrors) {
      if (err.status === 412) {
        logger.warn({ err, ...verboseLogCtx(req) }, `Precondition Failed: ${err.token}`);
      } else if (isLoginError(err) || isInvalidJWTError(err) || isExpiredInvite(err)) {
        logger.info({ err, ...verboseLogCtx(req) }, err.message);
      } else {
        logger.error({ err, ...verboseLogCtx(req) }, 'api error handler');
      }
    } else if (!err.status) {
      // log unhandled api errors as these shoul usually be properly caught errors
      logger.error({ err, ...verboseLogCtx(req) }, 'Api error without status. Possible uncaught error');
    }
    res.status(err.status || 500).json(formatError(err, { isProdEnv: config.isProdEnv, logger }));
  }
};
