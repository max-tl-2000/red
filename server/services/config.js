/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

import pick from 'lodash/pick';
import config from '../config';
import { getPropertyId } from './properties';
import { ServiceError } from '../common/errors';
import { getMarketRentRange } from '../dal/inventoryRepo';
import { admin } from '../common/schemaConstants';
import { getPropertyTimezone } from '../dal/propertyRepo';
import { getPartySettings } from './party-settings';
import { getTenantSettings } from './tenantService';

const getConfig = async ctx => {
  if (!ctx.tenantId) {
    throw new ServiceError({
      token: 'INVALID_TENANT',
      status: 400,
    });
  }

  let tenantSettings = await getTenantSettings(ctx);
  if (tenantSettings) {
    tenantSettings = pick(tenantSettings, ['preferences']);
  }

  const cfg = {
    emailDomain: config.mail.emailDomain,
    smsTemplateNameMap: config.smsTemplateNameMap,
    pagePaths: config.pagePaths,
    tenantName: ctx.tenantName,
    tenantSettings,
    // example of accessing the translator context
    // in the server
    //
    // ctx.i18n.translator.translate === i18next.t in the client
    // appTitle: ctx.i18n.translator.translate('APP_TITLE'),
  };

  if (ctx.tenantId !== admin.id) {
    const [marketRentRange, partySettings] = await Promise.all([getMarketRentRange(ctx.tenantId), getPartySettings(ctx)]);
    cfg.marketRentRange = marketRentRange;
    cfg.partySettings = partySettings;
  }

  return cfg;
};

export const getTenantAndPropertyIds = async (ctx, propertyName) => {
  let temporalPropertyName = propertyName;
  if (propertyName.toLowerCase() === 'parkmerced') {
    temporalPropertyName = 'swparkme';
  } // TODO: This code is to switch from Parkmerced to swparkme in order to find the id

  const propertyId = await getPropertyId(ctx, temporalPropertyName);
  if (!propertyId) {
    throw new ServiceError({
      token: 'INVALID_PROPERTY_FOR_GIVEN_TENANT',
      status: 400,
    });
  }

  const propertyTimezone = await getPropertyTimezone(ctx, propertyId);
  return {
    tenantId: ctx.tenantId,
    tenantName: ctx.tenantName,
    propertyName: propertyName === 'swparkme' ? 'Parkmerced' : propertyName, // TODO: This code is to switch from swparkme to Parkmerced in order to display Parkmerced
    propertyId,
    propertyTimezone,
  };
};

export { getConfig };
