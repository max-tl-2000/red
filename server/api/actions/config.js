/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

import * as service from '../../services/config';
import { ServiceError } from '../../common/errors';

export const getConfig = request => service.getConfig(request);

export const getTenantAndPropertyIds = request => {
  const { propertyName, tenant } = request.query;

  if (!tenant || !request.tenantId) {
    throw new ServiceError({ token: 'INVALID_TENANT', status: 400 });
  }

  if (!propertyName) {
    throw new ServiceError({ token: 'INVALID_PROPERTY', status: 400 });
  }

  return service.getTenantAndPropertyIds(request, propertyName);
};
