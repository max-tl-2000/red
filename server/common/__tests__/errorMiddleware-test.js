/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

import config from '../../config';
import { errorHandler } from '../errorMiddleware';
import logger from '../../../common/helpers/logger';

import * as overrider from '../../../common/test-helpers/overrider';

const createMockResponse = () => ({
  json: jest.fn(),
  status: jest.fn().mockReturnThis(),
  redirect: jest.fn(),
});

describe('API/error handling middleware', () => {
  beforeEach(() => {
    const fn = jest.fn();
    const mockLogger = { error: fn, info: fn, debug: fn, warn: fn };
    overrider.override(logger, mockLogger);
  });

  afterEach(() => overrider.restore());

  describe('when error is String', () => {
    it('has status code 500', () => {
      const res = createMockResponse();
      errorHandler()('some error', {}, res, () => {});

      expect(res.status).toBeCalledWith(500);
    });

    it('has JSON string in body', () => {
      const res = createMockResponse();
      errorHandler()('some error', {}, res);

      expect(res.json).toBeCalledWith('some error');
    });
  });

  describe('when error is Object', () => {
    it('redirects if error has redirect location', () => {
      const error = {
        redirect: '/foo/bar',
      };
      const res = createMockResponse();
      errorHandler()(error, {}, res);

      expect(res.redirect).toBeCalledWith(error.redirect);
    });

    it('has status code 500 if other not specified', () => {
      const error = {};

      const res = createMockResponse();
      errorHandler()(error, {}, res);

      expect(res.status).toBeCalledWith(500);
    });

    it('has custom status code if specified', () => {
      const error = {
        status: 429,
      };

      const res = createMockResponse();
      errorHandler()(error, {}, res);

      expect(res.status).toBeCalledWith(error.status);
    });

    it('has JSON error object in body', () => {
      const error = {
        foo: 'bar',
      };

      const res = createMockResponse();
      errorHandler()(error, {}, res);

      expect(res.json).toBeCalledWith(error);
    });
  });

  describe('when error is Error', () => {
    it('redirects if error has redirect location', () => {
      const error = Object.assign(new Error(), { redirect: '/foo/bar' });
      const res = createMockResponse();
      errorHandler()(error, {}, res);

      expect(res.redirect).toBeCalledWith(error.redirect);
    });

    it('has status code 500 if other not specified', () => {
      const error = Object.assign(new Error(), {});
      const res = createMockResponse();
      errorHandler()(error, {}, res);

      expect(res.status).toBeCalledWith(500);
    });

    it('has custom status code if specified', () => {
      const error = Object.assign(new Error(), { status: 429 });
      const res = createMockResponse();
      errorHandler()(error, {}, res);

      expect(res.status).toBeCalledWith(error.status);
    });

    it('has JSON error object in body', () => {
      overrider.override(config, {
        isProdEnv: true,
      });

      const error = Object.assign(new Error(), {
        foo: 'bar',
        baz: 'moo',
      });
      const res = createMockResponse();
      errorHandler()(error, {}, res);

      const result = res.json.mock.calls[0][0];

      expect(result.foo).toBeUndefined();
      expect(result.baz).toBeUndefined();

      overrider.restore();
    });

    it('In dev env the error contain the stack info', () => {
      overrider.override(config, {
        isProdEnv: false,
      });
      const error = new Error();

      const res = createMockResponse();
      errorHandler()(error, {}, res);

      const result = res.json.mock.calls[0][0];

      expect(result.stack).toBeDefined();

      overrider.restore();
    });

    it('has no error stacktrace in body if in PROD mode', () => {
      overrider.override(config, {
        isProdEnv: true,
      });
      const error = new Error();

      const res = createMockResponse();
      errorHandler()(error, {}, res);

      const result = res.json.mock.calls[0][0];

      expect(result.stack).not.toBeDefined();

      overrider.restore();
    });

    it('does not filter the data property in prod', () => {
      overrider.override(config, {
        isProdEnv: true,
      });
      const data = { foo: 'foo', bar: 'bar' };
      const error = Object.assign(new Error(), { data });

      const res = createMockResponse();
      errorHandler()(error, {}, res);

      const result = res.json.mock.calls[0][0];

      expect(result.data).toEqual(data);

      overrider.restore();
    });

    it('should not filter the data property in dev', () => {
      overrider.override(config, {
        isProdEnv: false,
      });
      const data = { foo: 'foo', bar: 'bar' };
      const error = Object.assign(new Error(), { data });

      const res = createMockResponse();
      errorHandler()(error, {}, res);

      const result = res.json.mock.calls[0][0];

      expect(result.data).toEqual(data);

      overrider.restore();
    });
  });

  it('should handle JsonSchemaValidation errors differently than normal errors', () => {
    const error = Object.assign(new Error(), {
      name: 'JsonSchemaValidation',
      validations: {
        body: [
          {
            messages: ['is not one of enum values: prospect,bronze,silver,gold'],
            property: 'request.body.score',
            value: 'violet',
          },
        ],
      },
    });

    const res = createMockResponse();
    errorHandler()(error, {}, res);

    const result = res.json.mock.calls[0][0];

    expect(result).toMatchSnapshot();
  });

  describe('when config.logMiddlewareErrors is false', () => {
    describe('and no status is present in the error', () => {
      it('should default to erro 500', () => {
        overrider.override(config, {
          logMiddlewareErrors: false,
        });

        const error = new Error();

        const res = createMockResponse();
        errorHandler()(error, {}, res);
        expect(res.status).toBeCalledWith(500);

        overrider.restore();
      });
    });

    it('should call status with the error.status', () => {
      overrider.override(config, {
        logMiddlewareErrors: false,
      });

      const error = Object.assign(new Error(), { status: 430 });

      const res = createMockResponse();
      errorHandler()(error, {}, res);
      expect(res.status).toBeCalledWith(430);

      overrider.restore();
    });
  });

  describe('when config.logMiddlewareErrors is true', () => {
    it('should log 412 errors as warnings', () => {
      overrider.override(config, {
        logMiddlewareErrors: true,
      });

      const error = Object.assign(new Error(), { status: 412, token: 'SOME_TOKEN' });

      const res = createMockResponse();
      errorHandler()(error, {}, res);
      expect(res.status).toBeCalledWith(412);

      const warnLog = logger.warn.mock.calls[0][1];
      expect(warnLog).toMatchSnapshot();

      overrider.restore();
    });

    it('should log authorization errors as info', () => {
      overrider.override(config, {
        logMiddlewareErrors: true,
      });

      const error = Object.assign(new Error('ACCOUNT_BLOCKED'), { status: 401, token: 'ACCOUNT_BLOCKED' });

      const res = createMockResponse();
      errorHandler()(error, {}, res);
      expect(res.status).toBeCalledWith(401);

      const infoLog = logger.info.mock.calls[0][1];
      expect(infoLog).toMatchSnapshot();

      overrider.restore();
    });

    it('should log InvalidJWTError errors as info', () => {
      overrider.override(config, {
        logMiddlewareErrors: true,
      });

      const error = Object.assign(new Error('INVALID_TOKEN'), { status: 401, code: 'INVALID_TOKEN' });

      const res = createMockResponse();
      errorHandler()(error, {}, res);
      expect(res.status).toBeCalledWith(401);

      const infoLog = logger.info.mock.calls[0][1];
      expect(infoLog).toMatchSnapshot();

      overrider.restore();
    });

    it('should log errors with status code 498 as info', () => {
      overrider.override(config, {
        logMiddlewareErrors: true,
      });

      const error = Object.assign(new Error('INVALID_TOKEN'), { status: 498, message: 'INVALID_TOKEN' });

      const res = createMockResponse();
      errorHandler()(error, {}, res);
      expect(res.status).toBeCalledWith(498);

      const infoLog = logger.info.mock.calls[0][1];
      expect(infoLog).toMatchSnapshot();

      overrider.restore();
    });
  });
});
