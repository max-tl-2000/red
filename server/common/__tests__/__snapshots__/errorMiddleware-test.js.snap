// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`API/error handling middleware should handle JsonSchemaValidation errors differently than normal errors 1`] = `
Object {
  "jsonSchemaValidation": true,
  "statusText": "Bad Request",
  "validations": Object {
    "body": Array [
      Object {
        "messages": Array [
          "is not one of enum values: prospect,bronze,silver,gold",
        ],
        "property": "request.body.score",
        "value": "violet",
      },
    ],
  },
}
`;

exports[`API/error handling middleware when config.logMiddlewareErrors is true should log 412 errors as warnings 1`] = `"Precondition Failed: SOME_TOKEN"`;

exports[`API/error handling middleware when config.logMiddlewareErrors is true should log InvalidJWTError errors as info 1`] = `"INVALID_TOKEN"`;

exports[`API/error handling middleware when config.logMiddlewareErrors is true should log authorization errors as info 1`] = `"ACCOUNT_BLOCKED"`;

exports[`API/error handling middleware when config.logMiddlewareErrors is true should log errors with status code 498 as info 1`] = `"INVALID_TOKEN"`;
